src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs

```diff
@@ -56,7 +63,17 @@ public override T[] Rent(int minimumLength)
 
             // Gettings here means we have too big of a request OR all the buckets from 
             // index through _buckets.Length are taken so we need to allocate a buffer on-demand.
-            return new T[Utilities.GetMaxSizeForBucket(index)];
+            int maxLength = Utilities.GetMaxSizeForBucket(_buckets.Length);
+            buffer = new T[Utilities.GetMaxSizeForBucket(index)];
+            if (log.IsEnabled())
+            {
+                if (buffer.Length > maxLength) 
+                    log.BufferOverMaximumAllocated(buffer.GetHashCode(), buffer.Length, maxLength, GetHashCode());
+                else
+                    log.UnPooledBufferAllocated(buffer.GetHashCode(), buffer.Length, GetHashCode());
```

Comment:
Similar to my previous comment about PoolId, you might add a method:
```C#
private static int GetBufferId(T[] buffer) { return buffer.GetHashCode(); }
```
Same goes for any other GetHashCode usage for id purposes.